portfolio/src/main/webapp/script.js

```diff
@@ -31,3 +31,13 @@ function addRandomFact() {
   factContainer.innerText.style.color = "#FEE9E1";
   factContainer.innerText.style.font = "20px georgia";
 }
+
+
+/**
+ * Gets a hard-coded "Hello Fatima" greeting from the server and adds it to the DOM
+ */
+async function getGreetingUsingAsyncAwait() { 
+  const response = await fetch('/data');
+  const greeting = await response.text();
+  document.getElementById('greeting-container').innerHTML = greeting;
```

Comment:
Copying a comment from Brian on Andrew's PR that used innerHTML (is that something they intentionally told you to do?)

```Using innerHTML is inherently unsafe from a security point of view, because it's possible for text crafted by a malicious user of your site to end up being parsed as HTML, which would let them take control over the site. It's called a cross-site scripting (XSS) attack, and the best way to prevent it is to construct your DOM with createElement instead.```
You could also choose to use jQuery which ```has some great APIs for quickly constructing DOM hierarchies that are almost as convenient as using innerHTML without being evil. 🙂```

That all said, could you use innerText here, if greeting is just pure text?

Reply:
Noted! I think the sample files used .innerHTML with createElement and we (Andrew and I) may have thought that was necessary here. I was also having some issues with my JS files not updating when running my webpage, and I think that setting the innerText seemed to temporarily resolve them. 
It seems that GitHub doesn't let me change this code here, so I'll create a new pull request on this branch with the .innerText update.

Reply:
Ah, I thought I had made this comment already! (Github has a learning curve for sure)

No problem. I'm sad they gave you bad starter code :(